woocommerce/woocommerce
# [Experimental] Product Filters Redesign > Filters: Implement Dynamic Block Title and Icon

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #48637 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate to Product Catalog template (Appearance > Editor > Templates > Product Catalog template - or any archive template).
2. Click on "Edit" to open the WordPress editor.
3. In the editor, you should see a block inserter icon or an option to "Add Block". Click on it.
4. Search for "Product Filters (Experimental)" in the block inserter or browse the available blocks until you find them. Add it to your content.
5. Open the List View and make sure the Product Filters (Experimental) contains the correct inner blocks: Filters, Active (Experimental), Price (Experimental), Status (Experimental), Color (Experimental), and Rating (Experimental). 
6. Make sure that each inner block contain their own icon according to the design.

The "Color (Experimental)" block should show if you're using the default WooCommerce sample data products. If you have more products, the block name should be the attribute with the highest product count on your store.

| Before | Current |
|--------|--------|
| ![CleanShot 2024-06-19 at 14 39 57@2x](https://github.com/woocommerce/woocommerce/assets/20469356/fb5ce7e0-e98a-4476-a1a1-97dd86e2aabf) | ![CleanShot 2024-06-21 at 17 53 31@2x](https://github.com/woocommerce/woocommerce/assets/20469356/efe7743e-2783-4b50-acad-1f88569d7f23) | 



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment  This PR does not require a changelog as the changes are behind an experimental flag and will not affect users.

</details>
